### DISCUSSION TITLE
GCP deployment - Module not found: Can't resolve '@public/badge/badge-xxl-execellent.png', for png/svg file

### DISCUSSION THREAD

**Discussion by krishna-fenixwork** [vercel/next.js#48965] [Help] [answered]
### Summary

Hey, everyone! Can someone help me with this, I've been stuck with this error since morning, couldn't understand what is happening. I'm deploying my next js application through gcp cloud build and Firebase, and for some reason, build fails and logs the error like this for images like png/svg - 

```Module not found: Can't resolve '@public/badge/badge-xxl-execellent.png'```

![image](https://user-images.githubusercontent.com/78067517/235216019-1a298702-1f43-44c6-95bd-bf28c3267e25.png)
![image](https://user-images.githubusercontent.com/78067517/235217153-e0812ed8-27eb-43e5-889a-1d2f5adf0b84.png)

Even though I have all these images inside my public folder and all works fine in local server. This error comes while deployment which makes the deployment fails. 

The log message also states - ``` Build failed because of webpack errors; Error ID: 1a2262f3 ```, and I don't have such webpack configuration in my project.

Here is my ```next.config.js``` file - 

```
/** @type {import('next').NextConfig} */

module.exports = {
    reactStrictMode: true,
    swcMinify: false,
    distDir: ".next",
    images: {
        domains: [
            "storage.googleapis.com",
            "firebasestorage.googleapis.com",
            "crunchbase-production-res.cloudinary.com",
            "lh3.googleusercontent.com",
        ],
    },
};

```

That's the ```tsconfig.json``` - 

```
{
	"compilerOptions": {
		"target": "es6",
		"lib": ["dom", "dom.iterable", "esnext"],
		"allowJs": true,
		"skipLibCheck": true,
		"strict": true,
		"forceConsistentCasingInFileNames": true,
		"noEmit": true,
		"esModuleInterop": true,
		"module": "esnext",
		"moduleResolution": "node",
		"resolveJsonModule": true,
		"isolatedModules": true,
		"jsx": "preserve",
		"incremental": true,
		"baseUrl": ".",        
		"paths": {
			"@components/*": ["components/*"],
			"@styles/*": ["styles/*"],
			"@public/*": ["public/*"],
			"@services/*": ["services/*"],
			"@config/*": ["config/*"],
			"@HOC/*": ["HOC/*"],
			"@Context/*": ["Context/*"],
			"@api/*": ["api/*"]
		}
	},
	"include": [
		"next-env.d.ts",
		"**/*.ts",
		"**/*.tsx",
		"Context/UserContext.js",
		"global.d.ts",
		"config/admin-config.js"
	],
	"exclude": ["./node_modules"]
}
```

I'm using the latest next.js and react.js version - 

```
"next": "^13.1.2",    
"react": "^18.2.0",   
"react-dom": "^18.2.0",    
"react-router-dom": "^6.0.1",


// dev dependencies
 "devDependencies": {
        "@types/node": "16.11.10",
        "@types/react": "^18.0.18",
        "@types/react-dom": "^18.0.6",
        "eslint": "8.20.0",
        "eslint-config-next": "^13.1.2"
    },
```

Please can someone help me on this and make me understand why it is happening?

**Comment by vladikk** [comment, marked as answer]
There was a recent breaking change in Cloud Functions that probably caused this issue:
https://cloud.google.com/functions/docs/release-notes#April_11_2023

**Reply by krishna-fenixwork** [reply]
@vladikk Thanks for the response. I followed the link and added `gcp-build:""` on my package.json scripts and tried to deploy and it worked! 🎉
